.github/workflows/tag-chart.yaml

```diff
@@ -0,0 +1,120 @@
+# Reusable workflow to tag a single chart and push it to a helm registry
+name: Tag Chart
+run-name: Tag ${{ inputs.bump }} version of ${{ inputs.chart }} by @${{ github.actor }}
+
+permissions:
+  contents: write
+
+on:
+  workflow_call:
+    inputs:
+      chart:
+        description: Name of chart to tag
+        required: true
+        type: string
+      bump:
+        description: "Bump version"
+        required: true
+        default: patch
+        type: string
+      HELM_REPO_BASE_URL:
+        type: string
+        description: Helm repository base url
+        required: true
+
+    secrets:
+      WORKFLOW_ACCESS_TOKEN:
+        description: "A GitHub PAT token with permissions to push to master branch"
+        required: true
+      HELM_REPO_USERNAME:
+        description: "Robot account user for the helm registry"
+        required: true
+      HELM_REPO_PASSWORD:
+        description: "Password for the helm registry"
+        required: true
+
+    outputs:
+      new_version:
+        value: ${{ jobs.publish_chart.outputs.new_version }}
+
+jobs:
+  publish_chart:
+    runs-on: ubuntu-latest
+    outputs:
+      new_version: ${{ steps.version.outputs.new_version }}
+      new_tag: ${{ steps.version.outputs.new_tag }}
+      last_tag: ${{ steps.version.outputs.last_tag }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false # To use WORKFLOW_ACCESS_TOKEN instead of GITHUB_TOKEN in later steps
```

Comment:
Can you explain this line and its comment?

Reply:
Every workflow in GitHub actions is provided a token by default stored in `$GITHUB_TOKEN` which is a built in service account for every repository, actions made on the repo using this token appear as done by `github-actions`.

This token however does not have permissions in the repository to push to protected branches and other actions that require admin privileges. We need to update the version with a commit on master which is protected, and push a tag.

To overcome this, we have a GitHub Personal Access Token (PAT) configured in the swan-cern organisation level secrets, with the token of a user with admin privileges (Diogo) which is stored in the secret WORKFLOW_ACCESS_TOKEN.

The checkout step caches the credentials for git in the `.git/` folder. In a later step where we want to use `WORKFLOW_ACCESS_TOKEN`, the cached `GITHUB_TOKEN` takes precedence, so we need to disable this behaviour